kratos/processes/levelset_convection_process.h

```diff
@@ -181,7 +202,7 @@ class LevelSetConvectionProcess
     ///@name Operations
     ///@{
 
-    void Execute() override
+    void ExecutePartially(const double dt_factor)
```

Comment:
This is breaking the process API. I think you could achieve the same by using an auxiliary `private` function.

Reply:
Please note that there would be two calls, one marching with dt_factor*DT and another one with (1 - dt_factor)*DT. So, I can not have dt_factor as an initialized variable. I mean there should be some way to pass "dt_factor" in the python solver. In the future, maybe, we need to modify the splitting algorithm.

Reply:
You must pass it by constructor. The `Process` API cannot be modified.

Reply:
Then, how can I recognize if this is the second step (which is marching with (1 - dt_factor)*DT) ?
This not problematic for dt_factor = 0.5, but what about for example, dt_factor = 0.3 ?

Reply:
If you agree, I will add **DELTA_TIME_FACTOR** so that it is easy to perform splitting, just set it before calling Execute.
It might also be useful for others.

Reply:
@mrhashemi the interface of processes is prescribed by the baseclass, so you cannot simply add a method (read, you shouldn't).

The way is to have the time_factor to be provided in the constructor

Reply:
I assume that I can have **DELTA_TIME_FACTOR** in the ProcessInfo and circumvent the issue.

Reply:
By why not passing it by constructor As we propose? If you pass it by the `ProcessInfo` you cannot provide defaults. This means that you cannot ensure it is always provided.

Reply:
I have added DELTA_TIME_FACTOR and removed ExecutePartially to comply with the rules. 
I should also add a default value (which can be passed by constructor); I will do it right away.
However, we can discuss about the special cases (where we need access to DELTA_TIME_FACTOR) later.

Reply:
I think that now I completely understand your problem. If you need to "partially" convect according to the `dt_factor` provided by constructor, you should use the `InitializeSolutionStep` and `FinalizeSolutionStep`. I don't see how we can do partial convect without violating the `Execute` (indeed I don't think the `Execute` is the correct function if we are not doing the convection completely).

Reply:
This should be solved with the new `partial_convection` entry of the settings. If this is active, the process would check the process info for a `DELTA_TIME_FACTOR`. If this is not provided, an error is thrown. From the user perspective, only the delta time factor must be updated in the solver according to the splitting scheme.